pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -976,7 +980,11 @@ public void deleteCursorFailed(ManagedLedgerException exception, Object ctx) {
     }
 
     void removeSubscription(String subscriptionName) {
-        subscriptions.remove(subscriptionName);
+        PersistentSubscription sub = subscriptions.remove(subscriptionName);
+        // preserve accumulative stats form removed subscription
+        SubscriptionStats stats = sub.getStats(false, false);
```

Comment:
Do you think it'd be worth adding specific getters for the two counters we want here? I am specifically concerned about the overhead of building out the whole `SubscriptionStats` object many times. I'm not sure _how_ expensive the `getStats` method really is, though.

Reply:
`removeSubscription` does not happen that frequently.

Reply:
Oh, right. I had been thinking of consumers, not subscriptions.